src/Npgsql/PgPassEntry.cs

```diff
@@ -0,0 +1,103 @@
+
+namespace Npgsql
+{
+    /// <summary>
+    /// Represents a hostname, port, database, username, and password combination that has been retrieved from a .pgpass file
+    /// </summary>
+    class PgPassEntry
+    {
+        #region Fields and Properties
+
+        /// <summary>
+        /// Hostname parsed from the .pgpass file
+        /// </summary>
+        internal string Hostname { get; }
+        /// <summary>
+        /// Port parsed from the .pgpass file
+        /// </summary>
+        internal string Port { get; }
+        /// <summary>
+        /// Database parsed from the .pgpass file
+        /// </summary>
+        internal string Database { get; }
+        /// <summary>
+        /// User name parsed from the .pgpass file
+        /// </summary>
+        internal string UserName { get; }
+        /// <summary>
+        /// Password parsed from the .pgpass file
+        /// </summary>
+        internal string Password { get; }
+
+        #endregion
+
+        #region Construction / Initialization
+
+        /// <summary>
+        /// This class represents an entry from the .pgpass file
+        /// </summary>
+        /// <param name="hostname">Hostname parsed from the .pgpass file</param>
+        /// <param name="port">Port parsed from the .pgpass file</param>
+        /// <param name="database">Database parsed from the .pgpass file</param>
+        /// <param name="userName">User name parsed from the .pgpass file</param>
+        /// <param name="password">Password parsed from the .pgpass file</param>
+        PgPassEntry(string hostname, string port, string database, string userName, string password)
+        {
+            Hostname = hostname;
+            Port = port;
+            Database = database;
+            UserName = userName;
+            Password = password;
+        }
+
+        /// <summary>
+        /// Creates new <see cref="PgPassEntry"/> based on string in the format hostname:port:database:username:password. The : and \ characters should be escaped with a \.
+        /// </summary>
+        /// <param name="serializedEntry">string for the entry from the pgpass file</param>
+        /// <returns>New instance of <see cref="PgPassEntry"/> for the string</returns>
+        /// <exception cref="FormatException">Entry is not formatted as hostname:port:database:username:password or non-wildcard port is not a number</exception>
+        internal static PgPassEntry Create(string serializedEntry)
+        {
+            var parts = Regex.Split(serializedEntry, @"(?<!\\):"); // split on any colons that aren't preceded by a \ (\ indicates that the colon is part of the content and not a separator)
+            if (parts == null || parts.Length != 5)
+            {
+                throw new FormatException(".PGPASS entry was not well-formed. Please ensure all non-comment entries are formatted as hostname:port:database:username:password. If colon is included, it must be escaped like \\:.");
```

Comment:
I did some testing with psql using a malformed pgpass file. It returns a message like this:

psql: FATAL:  password authentication failed for user "npgsql_tests"
password retrieved from file "_mypgpassfile_"

Based on that, I assume you want to leave the exception in place?